### PR TITLE
add multi-column index key part limit

### LINKED ISSUE #2273: A maximum of 16 columns is permitted for multicolumn indexes
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
 create table t(a char, b char, c char, d char, e char, f char, g char, h char ,i char, j char, k int, l char ,m char , n char, o char , p char, q char, index(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q));

2. What did you expect to see?

ERROR 1070 (42000): Too many key parts specified; max 16 parts allowed

3. What did you see instead?

Query OK, 0 rows affected (0.01 sec)

### PR DESCRIPTION
**Pull request by mxlxm**
fix #2273

### CODE REVIEW

**Comment by shenli** [comment]
@mxlxm Thanks!
LGTM

File: plan/validator.go [hunk 1]
```diff
@@ -384,6 +384,9 @@ func checkIndexInfo(indexName string, indexColNames []*ast.IndexColName) error {
 	if strings.EqualFold(indexName, mysql.PrimaryKeyName) {
 		return ddl.ErrWrongNameForIndex.GenByArgs(indexName)
 	}
+	if len(indexColNames) > 16 {
```
Comment (ngaut) [inline plan/validator.go]: Could you define the magic 16 as constant or add comments for the magic number?

**Comment by zz-jason** [comment]
@mxlxm CI failed.

File: plan/validator.go (hunk 1 above)
Comment (mxlxm) [inline plan/validator.go]: done.

**Comment by mxlxm** [comment]
@zz-jason last ci failed with `unionStore`'s `opts` field
maybe we should fix this in another pr.

**Pull request merged**